### ISSUE TITLE
Errors on service start on CLI are silenced

### ISSUE THREAD

**Issue by pedrudehuere** [zatosource/zato#432]
When starting a service in command line, errors are silenced.
 zato/code/zato-cli/src/zato/cli/start.py
 in start_component(...), run is called but no test is done wether the command run successfully or not.

**Comment by dsuch** [comment]
Hi @pedrudehuere - thanks for opening the ticket!

Can you please send in an example of the behaviour you find unsatisfactory + a few sentences on how you'd rather if it was working?

Cheers!

**Comment by pedrudehuere** [comment]
I followed the quick-start tutorial and created the cluster as illustrated in the tutorial.
Then I exeecuted `zato-qs-start.sh` in the directory where the cluster has been created and it sayd:
```
Running sanity checks
[1/6] Redis connection OK
[2/6] SQL ODB connection OK
[3/6] Load-balancer started
[4/6] server1 started
[5/6] server2 started
[6/6] Web admin started
Zato cluster quickstart-399470 started
Visit https://zato.io/support for more information and support options
```

But **the components didn't start**, I then executed the commands which are executed with the run() method by hand and found out that there was an error (the error was my fault).
I lost quite some time because of this. A simple check of the return value and an error message could be very useful here. somethink like:
```python
def start_component(...):
    ...
    if run(program, async=False if self.args.fg else True).returncode != 0:
        print_error()
    ...
```

Cheers!

**Comment by dsuch** [comment]
Ok, thanks, definitely makes sense.

However, could you tell me more about that misconfiguration? The thing is, there is the 'zato check-config' command that is being run before starting a quickstart environment:

https://zato.io/docs/2.0/admin/cli/check-config.html
https://github.com/zatosource/zato/blob/master/code/zato-cli/src/zato/cli/check_config.py

Perhaps that thing of yours could be added as well?

**Comment by pedrudehuere** [comment]
I use Debian 8 and because of [this](https://github.com/gevent/gevent/issues/513) problem I had to replace gevent 1.0 (which is the one installed by zato) with the development version. However, I did not do this for the `py` executable in zato/code/bin (which is used to run components) and that's why there was an error. I think the solution would be to upgrade to gevent 1.0.2 as soon as it is released (I hope soon).
Thanks.

**Comment by dsuch** [comment]
I see, right. Zato has not been tested with Debian 8 yet but will in some time after the OS is released.

Thanks for the heads up that we need to pay attention to gevent's version when Debian 8 is out.

The easiest way to update the dependencies is listed here:

https://mailman-mail5.webfaction.com/pipermail/zato-discuss/2015-February/000843.html

It takes care of updating all the paths without manual edits.

However, your note made me realize that these steps should actually be added to the docs.

In other words, double thanks :-)

**Comment by pedrudehuere** [comment]
Thanks for the link, it's a pleasure to contribute to awesome projects like zato =)

Cheers!

**Comment by dsuch** [comment]
Sure thing @pedrudehuere - it's great to have you here as well! :-)

**Comment by pedrudehuere** [comment]
Actually, I don't think this is a real issue, I could have simply run a server with --fg to see the error

**Comment by dsuch** [comment]
Ok, but if there is an error it should not be silenced out just like that because, until you spent a week or so with Zato, you didn't realize there was an --fg option I guess?

**Comment by pedrudehuere** [comment]
yes, that's right

**Comment by dsuch** [comment]
OK, I am working on it but this is less trivial than anticipated - there are subtle interactions and relations between capturing stdout/stderr while running Zato components in foreground or background mode, including if Python logging is used or not. In theory this is easy but in practice quite a bit of dabbling is needed.

For now I have tried [sarge](https://sarge.readthedocs.org/) and [executor](https://executor.readthedocs.org/). Neither of them provides a really convenient interface in this situation. I will keep working on it but it may turn out that in the end it will be simpler simply to add additional static checks to the list of what already runs before each `start` command.

Can you please explain your use case in detail again? Why did you need to modify `py` in the first place?

Thanks!

**Comment by dsuch** [comment]
Done in 10ba098cf46c082b772bed007213e65c0015cc51 - thanks @pedrudehuere !
